weasyl/comment.py

```diff
@@ -151,11 +154,14 @@ def insert(userid, submitid=None, charid=None, journalid=None, updateid=None, pa
         if not otherid:
             raise WeasylError("submissionRecordMissing")
     else:
-        # Determine otherid
-        otherid = d.execute("SELECT userid FROM %s WHERE %s = %i AND settings !~ 'h'",
-                            ["submission", "submitid", submitid] if submitid else
-                            ["character", "charid", charid] if charid else
-                            ["journal", "journalid", journalid], options="element")
+        # Determine the owner of the target
+        otherid = d.engine.scalar(
+            "SELECT userid FROM %s WHERE %s = %i AND settings !~ 'h'" % (
```

Comment:
I am pretty sure the input is all shoved through a ``define.get_int()`` before it gets here, but it feels bizarre doing a string format into here directly like this, since this doesn't have the same string escaping like ``d.execute`` did. There's at least one or two more instances of this elsewhere in this PR (if I recall correctly), but is it worth rewriting to ``%(id)s``/``id=submitid if submitid else charid if charid else journalid``, just for sanity's sake?

Just a thought.

Reply:
The real protection is `%i`, which only accepts integers – if `define.get_int` were forgotten, it’d throw. It’s equivalent to what `d.execute` would have done, since `d._sql_escape` didn’t change integers.